File: SUBS.py
def main():
    first_dna_string = input('Enter first DNA string (make sure all letters are UPPERCASE): ')
    second_dna_string = input('Enter second DNA string (make sure all letters are UPPERCASE): ')
    indexes = []
    index = 0
    if len(first_dna_string) < len(second_dna_string):
        print('Second DNA string must be equal or smaller than first DNA string')
    else:
        while index + len(second_dna_string) != len(first_dna_string) - 1:
            index = first_dna_string.find(second_dna_string, index)
            if index != -1:
                index += 1
                indexes.append(index)
            else:
                break
        print(indexes)


if __name__ == '__main__':
    main()

File: SUBS.java
package RosalindTasks;

import java.io.IOException;
import java.util.ArrayList;

public class SUBS {

	public static ArrayList<Integer> solveSUBSTask () throws IOException {
		var firstDataSet = Utils.readDataFromFile("resources/rosalind_subs.txt").get(0);
		var secondDataSet = Utils.readDataFromFile("resources/rosalind_subs.txt").get(1);
		int index = 0;
		ArrayList<Integer> indexes = new ArrayList<>();
		if (firstDataSet.length() < secondDataSet.length()) {
			System.out.println("Second DNA string must be equal or smaller than first DNA string");
		}
		else {
			while (index + secondDataSet.length() != firstDataSet.length()-1) {
				index = firstDataSet.indexOf(secondDataSet, index);
				if (index != -1) {
					index += 1;
					indexes.add(index);
				}
				else {
					break;
				}
			}
		}
		return indexes;
	}
	public static void main(String[] args) throws IOException {
		System.out.println(solveSUBSTask());
	}
}
